superset/assets/visualizations/nvd3_vis.js

```diff
@@ -624,13 +624,24 @@ function nvd3Vis(slice, payload) {
 
             const tip = tipFactory(e);
             const records = (slice.annotationData[e.name].records || []).map((r) => {
-              const timeColumn = new Date(r[e.timeColumn]);
+              const timeValue = new Date(r[e.timeColumn]);
               return {
                 ...r,
-                [e.timeColumn]: timeColumn,
+                [e.timeColumn]: timeValue,
               };
-            }).filter(r => !Number.isNaN(r[e.timeColumn].getMilliseconds()));
+            }).filter((r) => {
+              const isValid = !Number.isNaN(r[e.timeColumn].getMilliseconds());
+              if (isValid) {
+                xMin = xMin < r[e.timeColumn] ? xMin : r[e.timeColumn];
```

Comment:
I think its a bit weird that the filtering here has side effects. Might be cleaner to find the min and max values in that series after filtering.

Reply:
👍